doc/sphinx/language/gallina-extensions.rst

```diff
@@ -869,255 +870,297 @@ Module system
 The module system provides a way of packaging related elements
 together, as well as a means of massive abstraction.
 
-  .. productionlist:: modules
-    module_type       : `qualid`
-                      : `module_type` with Definition `qualid` := `term`
-                      : `module_type` with Module `qualid` := `qualid`
-                      : `qualid` `qualid` … `qualid`
-                      : !`qualid` `qualid` … `qualid`
-    module_binding    : ( [Import|Export] `ident` … `ident` : `module_type` )
-    module_bindings   : `module_binding` … `module_binding`
-    module_expression : `qualid` … `qualid`
-                      : !`qualid` … `qualid`
 
-  Syntax of modules
+.. cmd:: Module {? {| Import | Export } } @ident {* @module_binder } {? @of_module_type } {? := {+<+ @module_expr_inl } }
 
-In the syntax of module application, the ! prefix indicates that any
-`Inline` directive in the type of the functor arguments will be ignored
-(see the :cmd:`Module Type` command below).
+   .. insertprodn module_binder module_expr_inl
 
+   .. prodn::
+      module_binder ::= ( {? {| Import | Export } } {+ @ident } : @module_type_inl )
+      module_type_inl ::= ! @module_type
+      | @module_type {? @functor_app_annot }
+      functor_app_annot ::= [ inline at level @num ]
+      | [ no inline ]
+      module_type ::= @qualid
+      | ( @module_type )
+      | @module_type @module_expr_atom
+      | @module_type with @with_declaration
+      with_declaration ::= Definition @qualid {? @univ_decl } := @term
+      | Module @qualid := @qualid
+      module_expr_atom ::= @qualid
+      | ( {+ @module_expr_atom } )
+      of_module_type ::= : @module_type_inl
+      | {* <: @module_type_inl }
+      module_expr_inl ::= ! {+ @module_expr_atom }
+      | {+ @module_expr_atom } {? @functor_app_annot }
 
-.. cmd:: Module @ident
+   Defines a module named :token:`ident`.  See the examples :ref:`here<module_examples>`.
 
-   This command is used to start an interactive module named :token:`ident`.
+   The :n:`Import` and :n:`Export` flags specify whether the module should be automatically
+   imported or exported.
 
-.. cmdv:: Module @ident {* @module_binding}
+   Specifying :n:`{* @module_binder }` starts a functor with
+   parameters given by the :n:`@module_binder`\s.  (A *functor* is a function
+   from modules to modules.)
 
-   Starts an interactive functor with
-   parameters given by module_bindings.
+   .. todo: would like to find a better term than "interactive", not very descriptive
```

Comment:
I agree but I don't have any suggestion.

Reply:
I thought about referring to "single-command module definition" and "multi-command module definition" which is descriptive but also a mouthful.

Reply:
I wish the terminology was implying something about the use or not of the `:=` construct.